src/components/storage/MountPointMapping.jsx

```diff
@@ -486,31 +487,30 @@ const RequestsTable = ({
     deviceData,
     idPrefix,
     lockedLUKSDevices,
-    mountPointConstraints,
-    partitioningDataPath,
-    requests,
     setIsFormValid,
     setStepNotification,
 }) => {
-    const currentPartitioning = useRef();
-    const [unappliedRequests, setUnappliedRequests] = useState([]);
+    const mountPointConstraints = useMountPointConstraints();
+    const { partitioning } = useContext(StorageContext);
+    const requests = partitioning?.requests;
+    const reusePartitioning = useExistingPartitioning();
+    const [unappliedRequests, setUnappliedRequests] = useState();
     const allDevices = useMemo(() => {
         return requests?.filter(r => isUsableDevice(r["device-spec"], deviceData)).map(r => r["device-spec"]) || [];
     }, [requests, deviceData]);
+    const isLoadingPartitioning = !reusePartitioning || mountPointConstraints === undefined || !requests;
 
     // Add the required mount points to the initial requests
     useEffect(() => {
-        if (partitioningDataPath === currentPartitioning.current) {
+        if (isLoadingPartitioning || unappliedRequests !== undefined) {
             return;
         }
 
-        currentPartitioning.current = partitioningDataPath;
-
         const initialRequests = getInitialRequests(requests, mountPointConstraints);
         setUnappliedRequests(initialRequests);
 
         setIsFormValid(getRequestsValid(initialRequests, deviceData));
-    }, [deviceData, setIsFormValid, partitioningDataPath, requests, mountPointConstraints]);
+    }, [deviceData, setIsFormValid, partitioning.path, requests, isLoadingPartitioning, unappliedRequests, mountPointConstraints]);
```

Comment:
The `partitioning.path` dependency is listed here so that the hook is run in case applied partitioning changes, right?

Reply:
Correct, so if we have a new partitioning we need to recalculate the displayed requests.